variable_set.go

```diff
@@ -0,0 +1,274 @@
+	*Pagination
+	Items []*VariableSet
+}
+
+type VariableSet struct {
+	ID          string `jsonapi:"primary,varsets"`
+	Name        string `jsonapi:"attr,name"`
+	Description string `jsonapi:"attr,description"`
+	Global      bool   `jsonapi:"attr,global"`
+
+	// Relations
+	Organization *Organization          `jsonapi:"relation,organization"`
+	Workspaces   []*Workspace           `jsonapi:"relation,workspaces,omitempty"`
+	Variables    []*VariableSetVariable `jsonapi:"relation,vars,omitempty"`
+}
+
+// A list of relations to include. See available resources
+// https://www.terraform.io/docs/cloud/api/admin/organizations.html#available-related-resources
+type VariableSetIncludeOpt string
+
+const (
+	VariableSetWorkspaces VariableSetIncludeOpt = "workspaces"
+	VariableSetVars       VariableSetIncludeOpt = "vars"
+)
+
+type VariableSetListOptions struct {
+	ListOptions
+	Include string `url:"include"`
+}
+
+func (o VariableSetListOptions) valid() error {
+	return nil
+}
+
+// List all Variable Sets in the organization
+func (s *variableSets) List(ctx context.Context, organization string, options *VariableSetListOptions) (*VariableSetList, error) {
+	if !validStringID(&organization) {
+		return nil, ErrInvalidOrg
+	}
+	if options != nil {
+		if err := options.valid(); err != nil {
+			return nil, err
+		}
+	}
+
+	u := fmt.Sprintf("organizations/%s/varsets", url.QueryEscape(organization))
+	req, err := s.client.newRequest("GET", u, options)
+	if err != nil {
+		return nil, err
+	}
+
+	vl := &VariableSetList{}
+	err = s.client.do(ctx, req, vl)
+	if err != nil {
+		return nil, err
+	}
+
+	return vl, nil
+}
+
+// VariableSetCreateOptions represents the options for creating a new variable set within in a organization.
+type VariableSetCreateOptions struct {
+	// Type is a public field utilized by JSON:API to
+	// set the resource type via the field tag.
+	// It is not a user-defined value and does not need to be set.
+	// https://jsonapi.org/format/#crud-creating
+	Type string `jsonapi:"primary,vars"`
+
+	// The name of the variable set.
+	// Affects variable precedence when there are conflicts between Variable Sets
+	// https://www.terraform.io/cloud-docs/api-docs/variable-sets#apply-variable-set-to-workspaces
+	Name *string `jsonapi:"attr,name"`
+
+	// A description to provide context for the variable set.
+	Description *string `jsonapi:"attr,description,omitempty"`
+
+	// If true the variable set is considered in all runs in the organization.
+	Global *bool `jsonapi:"attr,global,omitempty"`
+}
+
+func (o VariableSetCreateOptions) valid() error {
+	if !validString(o.Name) {
+		return ErrRequiredName
+	}
+	if o.Global == nil {
+		return ErrRequiredGlobalFlag
+	}
+	return nil
+}
+
+// Create is used to create a new variable set.
+func (s *variableSets) Create(ctx context.Context, organization string, options *VariableSetCreateOptions) (*VariableSet, error) {
+	if !validStringID(&organization) {
+		return nil, ErrInvalidOrg
+	}
+	if options != nil {
```

Comment:
Can this check be moved to the valid() helper function? Like this: https://github.com/hashicorp/go-tfe/blob/20889e373e12cd0f20c35d7df3a37cfb711c7d70/admin_organization.go#L231